specs/APIReview_CustomDownload.md

```diff
@@ -0,0 +1,346 @@
+    }
+    ```
+
+    ```c# (but really MIDL3)
+    public class CoreWebView2NewWindowRequestedEventArgs
+    {
+        // ...
+
+	       public CoreWebView2WindowFeatures WindowFeatures { get; }
+    }
+    ```
+-->
+## Win32 C++
+```c#
+[uuid(9EAFB7D0-88C3-4450-BBFB-C05A46C40C72), object, pointer_default(unique)]
+interface ICoreWebView2Staging2 : IUnknown {
+
+  /// Add an event handler for the `DownloadStarting` event. This event fires
+  /// when a download has begun, blocking the default download UI.
+  ///
+  /// The host can choose to cancel a download, change the save path, and hide
+  /// the default download UI. If the host chooses to cancel the download, the
+  /// download is not saved and no UI is shown. Otherwise, the download is saved
+  /// once the event completes, and default download UI is shown if the host
+  /// did not choose to hide it.
+  ///
+  /// If the event is not handled, downloads will complete normally with the
+  /// default UI shown.
+  ///
+  HRESULT add_DownloadStarting(
+    [in] ICoreWebView2StagingDownloadStartingEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_DownloadStarting`.
+  HRESULT remove_DownloadStarting(
+      [in] EventRegistrationToken token);
+}
+
+[uuid(53b676ec-c3b1-4804-996a-c72c16b09efb), object, pointer_default(unique)]
+interface ICoreWebView2StagingDownloadStartingEventHandler : IUnknown
+{
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Staging2* sender,
+      [in] ICoreWebView2StagingDownloadStartingEventArgs* args);
+}
+
+/// Event args for the `DownloadStarting` event.
+[uuid(c7e95e2f-f789-44e6-b372-3141469240e4), object, pointer_default(unique)]
+interface ICoreWebView2StagingDownloadStartingEventArgs : IUnknown
+{
+  /// The uri of the download.
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The Content-Disposition header value from HTTP response.
+  [propget] HRESULT ContentDisposition([out, retval] LPWSTR* contentDisposition);
+
+  /// MIME type of the downloaded content.
+  [propget] HRESULT MimeType([out, retval] LPWSTR* mimeType);
+
+  /// The total number of expected bytes.
+  [propget] HRESULT TotalBytes([out, retval] UINT64* totalBytes);
+
+  /// The host may set this flag to cancel the download.
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+
+  /// Sets the `Cancel` property.
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// The save path of the download.
+  [propget] HRESULT SavePath([out, retval] LPWSTR* savePath);
+
+  /// Sets the `SavePath` property.
+  [propput] HRESULT SavePath([in] LPCWSTR savePath);
+
+  /// The host may set this flag to hide the default downloads UI.
+  [propget] HRESULT HideUI([out, retval] BOOL* hideUI);
+
+  /// Sets the `HideUI` property.
+  [propput] HRESULT HideUI([in] BOOL hideUI);
```

Comment:
Naming this "Hide..." makes it sound like a method. And 'UI' is not very specific. How about `ShouldDisplayDefaultDownloadDialog`?